### DISCUSSION TITLE
SEARCH method and Spring WebClient

### DISCUSSION THREAD

**Discussion by BenoitAverty** [directus/directus#15423] [Archive]
Hello,

I'm consuming a directus instance from a Kotlin/Spring application. My problem is that the WebClient class does not seem to have a way to perform a SEARCH http request, so I'm stuck. 

Has anyone encountered this problem and found a solution ? Without this I'll have to split my request into several requests with shorter URIs...

Thanks

**Comment by br41nslug** [comment]
It does indeed look like you're limited to these HTTP types https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/http/HttpMethod.html

That leaves you with 2 options:
- Use a more permissive web library
- Stick to `GET` requests (`SEARCH` is a nice addition but solves a pretty niche situation i.m.o)
